ddsrouter_core/include/ddsrouter_core/configuration/DDSRouterConfiguration.hpp

```diff
@@ -39,50 +39,39 @@ namespace configuration {
  * This class joins every DDSRouter feature configuration and includes methods
  * to interact with this configuration.
  */
-class DDSRouterConfiguration : public DDSRouterReloadConfiguration
+struct DDSRouterConfiguration : public DDSRouterReloadConfiguration
 {
-public:
 
-    /**
-     * TODO
-     */
+    /////////////////////////
+    // CONSTRUCTORS
+    /////////////////////////
+
+    DDSROUTER_CORE_DllAPI DDSRouterConfiguration() = default;
+
     DDSROUTER_CORE_DllAPI DDSRouterConfiguration(
             std::set<std::shared_ptr<types::FilterTopic>> allowlist,
             std::set<std::shared_ptr<types::FilterTopic>> blocklist,
             std::set<std::shared_ptr<types::RealTopic>> builtin_topics,
             std::set<std::shared_ptr<ParticipantConfiguration>> participants_configurations,
-            unsigned int number_of_threads = default_number_of_threads());
-
-    /**
-     * @brief Return a set with the different \c ParticipantConfigurations in the yaml
-     *
-     * Every participant configuration is an object of the specific class set in \c types::ParticipantKind .
-     *
-     * @return Set of \c ParticipantConfigurations
-     */
-    DDSROUTER_CORE_DllAPI std::set<std::shared_ptr<ParticipantConfiguration>> participants_configurations() const
-    noexcept;
+            unsigned int number_of_threads);
 
-    DDSROUTER_CORE_DllAPI bool is_valid(
-            utils::Formatter& error_msg) const noexcept override;
+    /////////////////////////
+    // METHODS
+    /////////////////////////
 
     DDSROUTER_CORE_DllAPI void reload(
             const DDSRouterReloadConfiguration& new_configuration);
 
-    DDSROUTER_CORE_DllAPI unsigned int number_of_threads() const noexcept;
-
-    DDSROUTER_CORE_DllAPI static unsigned int default_number_of_threads() noexcept;
-
-protected:
-
-    static bool check_correct_configuration_object_(
```

Comment:
Isn't this of any use?

Reply:
Shit. This was intended for another change: do Participants independent from configurations and use it as ptr and not as a templated variable. But I finally decided to do so in a forward PR.
The check is not so relevant anyway, it checks that the kind of the object actually belongs to the class, what only concern developers so far.
If you prefer it, I can added it again

Reply:
Yes please, just in case.